invenio_app_ils/circulation/utils.py

```diff
@@ -51,3 +53,29 @@ def circulation_is_loan_duration_valid(loan):
 def circulation_can_be_requested(loan):
     """Return True if the Document/Item for the given Loan can be requested."""
     return True
+
+
+def filter_circulation_status(circulation_status):
+    """Filter circulation status depending on user permissions."""
+    patron_pid = circulation_status.get("patron_pid", None)
+    if not patron_pid:
+        return circulation_status
+
+    filter_keys = [
+        "loan_pid",
+        "patron_pid",
+        "start_date",
+        "end_date",
+        "extension_count",
+        "request_expire_date",
+    ]
+
+    allowed = True
+    if has_request_context():
+        allowed = loan_owner_permission(patron_pid).can()
+
+    for key in filter_keys:
+        if key in circulation_status and not allowed:
+            del circulation_status[key]
```

Comment:
I would check if the key exists in the `circulation_status`, or maybe use `.pop()`. For example the `extension_count` is not there in some cases.